platforms/ios/framework/src/TGMapView.mm

```diff
@@ -1,44 +1,336 @@
+        _map->setDefaultBackgroundColor(red, green, blue);
+    }
+
+}
+
+- (void)setupDisplayLink
+{
+    BOOL visible = self.superview && self.window;
+    if (visible && !_displayLink) {
+        _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(displayLinkUpdate:)];
+        _displayLink.preferredFramesPerSecond = self.preferredFramesPerSecond;
+        [_displayLink addToRunLoop:[NSRunLoop currentRunLoop] forMode:NSRunLoopCommonModes];
+        _renderRequested = YES;
+    } else if (!visible && _displayLink) {
+        [_displayLink invalidate];
+        _displayLink = nil;
+    }
+}
+
+- (void)setupGestureRecognizers
+{
+    _tapGestureRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(respondToTapGesture:)];
+    _tapGestureRecognizer.numberOfTapsRequired = 1;
+    // TODO: Figure a way to have a delay set for it not to tap gesture not to wait long enough for a doubletap gesture to be recognized
+    _tapGestureRecognizer.delaysTouchesEnded = NO;
+
+    _doubleTapGestureRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(respondToDoubleTapGesture:)];
+    _doubleTapGestureRecognizer.numberOfTapsRequired = 2;
+    // Ignore single tap when double tap occurs
+    [_tapGestureRecognizer requireGestureRecognizerToFail:_doubleTapGestureRecognizer];
+
+    _panGestureRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(respondToPanGesture:)];
+    _panGestureRecognizer.maximumNumberOfTouches = 1;
+
+    _pinchGestureRecognizer = [[UIPinchGestureRecognizer alloc] initWithTarget:self action:@selector(respondToPinchGesture:)];
+    _rotationGestureRecognizer = [[UIRotationGestureRecognizer alloc] initWithTarget:self action:@selector(respondToRotationGesture:)];
+    _shoveGestureRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(respondToShoveGesture:)];
+    _shoveGestureRecognizer.minimumNumberOfTouches = 2;
+    _longPressGestureRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(respondToLongPressGesture:)];
+
+    // Use the delegate method 'shouldRecognizeSimultaneouslyWithGestureRecognizer' for gestures that can be concurrent
+    _panGestureRecognizer.delegate = self;
+    _pinchGestureRecognizer.delegate = self;
+    _rotationGestureRecognizer.delegate = self;
+
+    [_glView addGestureRecognizer:_tapGestureRecognizer];
+    [_glView addGestureRecognizer:_doubleTapGestureRecognizer];
+    [_glView addGestureRecognizer:_panGestureRecognizer];
+    [_glView addGestureRecognizer:_pinchGestureRecognizer];
+    [_glView addGestureRecognizer:_rotationGestureRecognizer];
+    [_glView addGestureRecognizer:_shoveGestureRecognizer];
+    [_glView addGestureRecognizer:_longPressGestureRecognizer];
+}
+
+- (void)didMoveToWindow
+{
+    [self setupDisplayLink];
+    [super didMoveToWindow];
+}
+
+- (void)didMoveToSuperview
```

Comment:
Should we also be handling `didAddSubview`?

Reply:
I can't think of anything we would need to do in `TGMapView` when a subview is added.